lib/matplotlib/axis.py

```diff
@@ -1529,37 +1529,68 @@ def set_major_formatter(self, formatter):
         """
         Set the formatter of the major ticker.
 
+        In addition to a `~matplotlib.ticker.Formatter` instance,
```

Comment:
As argued in the mailing list (not repro'ing the discussion here) I'd like to argue in favor of
```
This accepts either a `.Formatter` instance, or a callable, which is
automatically wrapped in a `.FuncFormatter`.  To use a format string
(e.g. "{x:.4f}"), pass the corresponding bound string method
(`"{x:.4f}".format`).
```
which I don't actually think is harder to understand (in one case the user needs to learn about StrMethodFormatter, which is a highly matplotlib specific concept; in the other case, about bound methods, which is also somewhat advanced, but at least much more widely applicable).

(Then you can remove all the mentions of StrMethodFormatter.)

Reply:
In my opinion the `StrMethodFormatter` is the most basic formatter.  There is probably a reason the documentation routinely uses the `FormatStrFormatter` rather than the `FuncFormatter`.  Further, in the documentation the string-related formatter are usually described first.  Learning to work with strings is one of the first thing new python developers encounter, while passing around function objects is much more advanced.

So although `StrMethodFormatter` is a new function, it isn't a new concept.  Once a beginning user sees they can use strings, they won't have to worry about functions in most cases.  Using a bound method like this, on the other hand, is a new concept for many users, one they wouldn't necessarily have to deal with otherwise.  This goes against the purpose of this change, which is to make things simpler and easier to use.

That is similar to the reason I intentionally don't even mention callables.  A beginning users isn't going to know what that is and is just going to get confused, while a more advanced user is going to understand that callables can be used wherever functions are normally used.

So although this makes the code simpler, it only really saves two lines.  And I think this comes at the expense of making things more difficult for beginning users, exactly the sort of people I was trying to help with this change.

Reply:
I agree with @toddrjen here.  We want to make this less arcane, not more arcane, and telling folks to pass the `format` method of a string is pretty arcane.   Most people want to just do: `format_my_ticks('%01.3f')`.  

WRT this, you apparently need the `x` in `'{x}'`?   This also needs to be documented?  If you are going to just refer people to the underlying Formatter classes to understand the syntax needed, then you lose a lot of the simplification here.

Reply:
I don't feel strongly enough to block over this, but enough to not approve :shrug: